packages/components/addon/components/hds/modal/index.js

```diff
@@ -156,7 +164,18 @@ export default class HdsModalIndexComponent extends Component {
   }
 
   @action
-  onDismiss() {
+  async onDismiss() {
+    // allow ember test helpers to be aware of when the `close` event fires
+    // when using `click` or other helpers from '@ember/test-helpers'
+    if (DEBUG && this.element.open) {
```

Comment:
With Ember-CLI's General Purpose ENV flags, `DEBUG` will be replaced with `false`, and this code will get stripped out in production builds by the minifier

Reply:
Checking if the element is open is required, otherwise we'll set up an event listener that will never fire if the modal is already closed.